miditoolkit/midi/containers.py

```diff
@@ -284,26 +284,18 @@ def __init__(
         self.control_changes = [] if control_changes is None else control_changes
         self.pedals = [] if pedals is None else pedals
 
-    def remove_invalid_notes(self, verbose: bool = True):
-        """Removes any notes whose end time is before or at their start time."""
-        # Crete a list of all invalid notes
-        notes_to_delete = [note for note in self.notes if note.end <= note.start]
-        if verbose:
-            if len(notes_to_delete):
-                print("\nInvalid notes:\n", notes_to_delete, "\n\n")  # noqa: T201
-            else:
-                print("no invalid notes found")  # noqa: T201
-            return True
-
-        # Remove the notes found
-        for note in notes_to_delete:
-            self.notes.remove(note)
-        return False
+    def remove_notes_with_no_duration(self):
+        """Removes (inplace) notes whose end time is before or at their start time."""
+        for i in range(self.nb_notes - 1, -1, -1):
+            if self.notes[i].start >= self.notes[i].end:
+                del self.notes[i]
```

Comment:
I'd keep the old name as a compatibility alias:

```suggestion
    def remove_invalid_notes(self, verbose: bool = True) -> None:
        warnings.warn("Call remove_notes_with_no_duration() instead.", DeprecationWarning)
        return self.remove_notes_with_no_duration()

    def remove_notes_with_no_duration(self) -> None:
        """Removes (inplace) notes whose end time is before or at their start time."""
        for i in range(self.nb_notes - 1, -1, -1):
            if self.notes[i].start >= self.notes[i].end:
                del self.notes[i]
```